packages/reporting/src/patterns.js

```diff
@@ -126,6 +141,53 @@ const TRANSFORMS = new Map(
       }
     },
 
+    /**
+     * Given a URL and a list of query parameters, it returns an equivalent
+     * URL, but with those query parameters removed.
+     *
+     * Notes:
+     * - If the parameter occurs multiple times, all of them will be removed.
+     * - If the URL is invalid, null is returned.
+     *
+     * Example ["removeParams", ["foo"]]:
+     * - "https://example.test/path?foo=remove&bar=keep" -> "https://example.test/path?bar=keep"
+     * - "This is a string but not an URL" -> null
+     * - "/example.test/path" -> null (relative URLs are not supported)
+     *
+     * Example ["removeParams", ["foo", "bar"]]:
+     * - "https://example.test/path?foo=1&bar=2" -> "https://example.test/path"
+     */
+    removeParams: (x, queryParams) => {
+      expectString(x);
+      expectArrayOfStrings(queryParams);
+      try {
+        const url = new URL(x);
+
+        // Why not url.searchParams? The parsing step in url.searchParams
+        // loses information (see https://stackoverflow.com/a/45516812/783510).
+        if (!url.search) {
```

Comment:
why not @cliqz/url-parser? does it also loose the information?

Reply:
I did some tests. It behaves slightly different with pros and cons:

```
const url = new cliqz.URL(x);
for (const param of queryParams) {
  url.searchParams.delete(param);
}
return url.toString();
```

For instance, it would change URLs like that:

`https://example.test/foo?bar`
->
`https://example.test/foo?bar=`

What it does do better (in the context of the purpose of this function) than the standard URL parser is that does not decode URL with non-ascii letter.

Will have to try out, but the more I think about it, doing everything by hand (extracting the search also) might be best (everything between `?` to the end or `#`). The decoding that the parser libraries are doing, is adding only extra complexity.

Reply:
The new implementation does not use any parser library, which gives more control and is arguably easier. Depending on the context, normalizing the URL could be useful, but IMHO it should be something to opt-in. For instance, there are the "decodeURIComponent" and "tryDecodeURIComponent" primitives already, which serve a similar purpose.

(I also extracted it, since the functionality to clean URLs from tracking parameters should be useful in the "page" context. But that is outside of the scope of this PR.)